File: ProjectFullstack/frontend/reactjs-app-tsx-loginn-2/reactjs-app-tsx/src/api/departmentService.ts
import api from "./axiosInstance";
import { Department } from "../types/entity";

const DEPARTMENTS_ENDPOINT = "/departments";

export const departmentService = {
  getAllDepartments: async (): Promise<Department[]> => {
    const response = await api.get<Department[]>(DEPARTMENTS_ENDPOINT);
    return response.data;
  },

  getDepartmentById: async (id: number): Promise<Department> => {
    const response = await api.get<Department>(`${DEPARTMENTS_ENDPOINT}/${id}`);
    return response.data;
  },

  getDepartmentsByPage: async (page: number, limit: number) => {
    const response = await api.get(`/departments/page`, {
      params: {
        page: page - 1,
        limit: limit,
      },
    });

    const pageData = response.data;

    return {
      data: pageData.content,
      total: pageData.totalElements,
    };
  },

  createDepartment: async (
    department: Omit<Department, "id">
  ): Promise<Department> => {
    const response = await api.post<Department>(
      DEPARTMENTS_ENDPOINT,
      department
    );
    return response.data;
  },

  updateDepartment: async (
    id: number,
    department: Omit<Department, "id">
  ): Promise<Department> => {
    const response = await api.put<Department>(
      `${DEPARTMENTS_ENDPOINT}/${id}`,
      department
    );
    return response.data;
  },

  deleteDepartment: async (id: number): Promise<void> => {
    await api.delete(`${DEPARTMENTS_ENDPOINT}/${id}`);
  },
};

export default departmentService;

File: ProjectFullstack/backend/Spring_Project_VTI-duong/Spring_Project_VTI-duong/spring_day11_1/spring/src/main/java/com/vti/springdatajpa/service/DepartmentService.java
package com.vti.springdatajpa.service;

import com.vti.springdatajpa.dto.AccountDto;
import com.vti.springdatajpa.dto.DepartmentDto;
import com.vti.springdatajpa.entity.DepartmentEntity;
import org.springframework.data.domain.Page;

import java.util.List;

public interface DepartmentService {
    public List<DepartmentEntity> getAllDepertment();

    public Page<DepartmentDto> getDepartmentsByPage(int page, int limit);

    public DepartmentEntity getDepartmentById(int id);

    public List<DepartmentEntity> getDepartmentByName(String name);

    public DepartmentDto createDepartment(DepartmentDto departmentDto);

    public void updateDepartment(DepartmentDto departmentDto, int id);

    public void deleteDepartment(int id);
}
